bindings/rust/s2n-tls/src/callbacks/session_ticket.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::time::Duration;
+
+use s2n_tls_sys::*;
+
+use crate::{
+    connection::Connection,
+    error::{Error, Fallible},
+};
+
+/// A trait to retrieve session tickets from the connection
+pub trait SessionTicketCallback: Send + Sync + 'static {
+    fn on_session_ticket(&self, connection: &mut Connection, session_ticket: &SessionTicket);
+}
+
+pub struct SessionTicket(s2n_session_ticket);
+
+impl SessionTicket {
+    pub(crate) fn from_ptr(ticket: &s2n_session_ticket) -> &Self {
+        unsafe { &*(ticket as *const s2n_session_ticket as *const SessionTicket) }
+    }
+
+    // SAFETY: casting *const s2n_session_ticket -> *mut s2n_session_ticket: This is
+    // safe as long as the data is not actually mutated. As authors of s2n-tls,
+    // we know that the get_lifetime and get_data methods do not mutate the
+    // data, and use mut pointers as a matter of convention because it makes
+    // working with s2n_stuffers and s2n_blobs easier.
+    pub(crate) fn deref_mut_ptr(&self) -> *mut s2n_session_ticket {
+        &self.0 as *const s2n_session_ticket as *mut s2n_session_ticket
+    }
+
+    pub fn lifetime(&self) -> Result<Duration, Error> {
```

Comment:
Is this an duration since unix epoch? I would recommend [SystemTime](https://doc.rust-lang.org/std/time/struct.SystemTime.html) instead, if so.

Reply:
It's not. It's just the number of seconds this ticket will be valid.[ You can see in the C function that generates the lifetime](https://github.com/aws/s2n-tls/blob/cb32a54dd95e38088b3873d4cc9b05bc4e8d77e3/tls/s2n_server_new_session_ticket.c#L178) that it's not sending "the duration since the unix epoch at which the ticket will no longer be valid."

Reply:
Ah ok. So it's a relative time to when the ticket was issues; basically its age?